test/e2e/specs/treeview/00-record-page.spec.js

```diff
@@ -0,0 +1,50 @@
+var httpFun = require('../../utils/common/httpFun.js');
+var deriva = require('../../utils/common/deriva-webapps');
+var testConfig = require('../../config');
+
+exports.tests = function (appName, baseUrl) {
+  describe('Record page '+ appName, function () {
+    var EC = protractor.ExpectedConditions;
+
+    beforeAll(function () {
+      browser.ignoreSynchronization=true;
+      browser.get(baseUrl+testConfig.treeviewPage.recordpage);
+      ele = element(by.id('rt-heading-Specimen&nbsp;Expression'));
+
+      deriva.waitForElementInverse(element.all(by.id("rt-loading")).get(0));
+      browser.wait(EC.visibilityOf(ele), browser.params.defaultTimeout);
+
+    });
+
+    it('verify that treeview are displayed', function () {
+      if (ele.isElementPresent(by.tagName('iframe'))) {
+        iframe = ele.element(by.tagName('iframe'));
+        iframe.getAttribute('src').then(function (iframeUrl) {
+          httpFun.fireHttpRequest(iframeUrl).then(function (res) {
+            expect(res.statusCode).toEqual(200);
+          }).catch(function (err) {
+            console.log("Error in promise chain: ", err);
+          });
+        });
```

Comment:
This should be changed to use promise chaining so that the `catch` block only needs to be defined once.
```js
iframe.getAttribute('src').then(function (iframeUrl) {
  return httpFun.fireHttpRequest(iframeUrl)
}).then(function (res) {
  expect(res.statusCode).toEqual(200);
}).catch(function (err) {
  console.log("Error in promise chain: ", err);
});
```